### PR TITLE
2017-03-01 api version of apim spec

### PR DESCRIPTION
**Pull request by solankisamir**
2017-03-01 Arm Compliant API Version which fixes issue
 - https://github.com/Azure/azure-rest-api-specs/issues/1060
 - https://github.com/Azure/azure-rest-api-specs/issues/893

This is an updated PR of the pull request https://github.com/Azure/azure-rest-api-specs/pull/1286

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by solankisamir** [comment]
@fearthecowboy I have opened a new PR for the closed PR https://github.com/Azure/azure-rest-api-specs/pull/1286, which you had reviewed. This includes fixes after generating the .net client.

**Comment by salameer** [comment]
Assigning to Garret since he already has context

**Comment by fearthecowboy** [comment]
ATTENTION REQUIRED.
-------------

We have changed the `default` branch in the `azure-rest-api-specs` repository to use the branch labeled `current`.

This PR is opened against the `master` branch.

You have two options: 
1. Keep this PR open, and complete the review process as normal, and once this PR is merged against the `master` branch, we (@olydis & @fearthecowboy)  will create a PR against the current branch and make any updates required to work in the newly reorganized default branch.  

2. Close this PR and create a new one. That would mean pulling in the changes from the new default branch `current` moving any files and switching to the configuration-based layout.  (You are more than welcome to choose option 2, but I don't expect anyone to do that)

Either way, the work you are currently doing against the previous default branch `master` will be the last PR against that branch that we will accept and all new PRs will go against the new default branch  `current`.

If you have any questions or concerns, ping @fearthecowboy or @olydis (or go ask on the `#swagger` channel in slack) ASAP and we can help you thru the process.

**Comment by solankisamir** [comment]
@fearthecowboy lets complete the review process in the master branch itself.

File: arm-apimanagement/2017-03-01/swagger/apimquotas.json [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+          "type": "string",
+          "format": "date-time",
+          "description": "The date of the start of Counter Period. The date conforms to the following format: `yyyy-MM-ddTHH:mm:ssZ` as specified by the ISO 8601 standard.\n"
+        },
+        "periodEndTime": {
+          "type": "string",
+          "format": "date-time",
+          "description": "The date of the end of Counter Period. The date conforms to the following format: `yyyy-MM-ddTHH:mm:ssZ` as specified by the ISO 8601 standard.\n"
+        },
+        "Value": {
```
Comment (fearthecowboy) [inline arm-apimanagement/2017-03-01/swagger/apimquotas.json]: `"Value"` should start with a lower-case  `v` to be camel-case consistent.

File: arm-apimanagement/2017-03-01/swagger/apimtenant.json [hunk 2]
```diff
@@ -0,0 +1,736 @@
...
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/AccessInformationUpdateParameters"
+            },
+            "description": "Parameters."
```
Comment (fearthecowboy) [inline arm-apimanagement/2017-03-01/swagger/apimtenant.json]: `"description" : "Parameters"` - Description must not match the name of the node it is supposed to describe.

File: arm-apimanagement/2017-03-01/swagger/apimdeployment.json [hunk 3]
```diff
@@ -0,0 +1,1055 @@
...
+            "Scm"
+          ],
+          "x-ms-enum": {
+            "name": "HostnameType",
+            "modelAsString": false
+          }
+        },
+        "hostName": {
+          "type": "string",
+          "description": "Hostname."
```
Comment (fearthecowboy) [inline arm-apimanagement/2017-03-01/swagger/apimdeployment.json]: `"description" : "Hostname"` - Description must not match the name of the node it is supposed to describe.

**Review by fearthecowboy — CHANGES_REQUESTED** [review]
I'm pretty sure we've talked about CreateOrUpdate before -- 

>  A PUT operation request body schema should be the same as its 200 response schema, to allow reusing the same entity between GET and PUT. 
> If the schema of the PUT request body is a superset of the GET response body, make sure you have a PATCH operation to make the resource updatable. 

In operations: 
``` 
    arm-apimanagement/2017-03-01/swagger/apimapis.json:149:6 
    arm-apimanagement/2017-03-01/swagger/apimcertificates.json:146:6
    arm-apimanagement/2017-03-01/swagger/apimemailtemplate.json:132:6
    arm-apimanagement/2017-03-01/swagger/apimgroups.json:143:6
    arm-apimanagement/2017-03-01/swagger/apimproducts.json:153:6 
    arm-apimanagement/2017-03-01/swagger/apimproperties.json:143:6
    arm-apimanagement/2017-03-01/swagger/apimsubscriptions.json:146:6
    arm-apimanagement/2017-03-01/swagger/apimusers.json:146:6
    arm-apimanagement/2017-03-01/swagger/apimopenidconnectproviders.json:140:6 
```    

You're not able to keep the models the same, because the creates have additional data? 

Does the API support round-tripping the GET entity with the PATCH operations?

-----

Operations missing `x-ms-examples` -- ( can these be added? ):

 'Api_Update'
 'Api_Delete'
 'ApiOperation_Update'
 'ApiOperationPolicy_CreateOrUpdate'
 'ApiPolicy_CreateOrUpdate'
 'ApiPolicy_Delete'
 'AuthorizationServer_Update'
 'Backend_Update'
 'Backend_Delete'
 'ApiManagementOperations_List'
 'EmailTemplate_Update'
 'Group_Update'
 'IdentityProvider_Update'
 'Logger_Update'
 'OpenIdConnectProvider_Update'
 'Product_Update'
 'Subscription_Update'
 'TenantConfiguration_Deploy'
 'TenantConfiguration_Save'
 'TenantConfiguration_Validate'
 'User_Update'


----

Other than that, I think this is pretty good. I've ignored the `R3018` , `R3010` and `R2061` linter errors, I don't think they are significant in this case.

File: arm-apimanagement/2017-03-01/swagger/apimquotas.json (hunk 1 above)
Comment (solankisamir) [inline arm-apimanagement/2017-03-01/swagger/apimquotas.json]: Yes.. this is an issue in the API. We are planning to fix it in the next Api-Version.

File: arm-apimanagement/2017-03-01/swagger/apimtenant.json (hunk 2 above)
Comment (solankisamir) [inline arm-apimanagement/2017-03-01/swagger/apimtenant.json]: Fixed

File: arm-apimanagement/2017-03-01/swagger/apimdeployment.json (hunk 3 above)
Comment (solankisamir) [inline arm-apimanagement/2017-03-01/swagger/apimdeployment.json]: Fixed

**Comment by solankisamir** [comment]
arm-apimanagement/2017-03-01/swagger/apimapis.json:149:6 
>> Yes the create parameter has extra parameters like Content-Type, which is not part of the Get Contract
arm-apimanagement/2017-03-01/swagger/apimcertificates.json:146:6
>> The Create Contract has Base64 Certificate, not part of the get Contract
    arm-apimanagement/2017-03-01/swagger/apimemailtemplate.json:132:6
>> isDefault is a Property in GetContract, not present in CreateContract
    arm-apimanagement/2017-03-01/swagger/apimgroups.json:143:6
>> Property builtiin is GetContract is not valid in CreateContract, as it is ReadOnly.
    arm-apimanagement/2017-03-01/swagger/apimproducts.json:153:6 
>> I have made the GetContract and CreateContract same
    arm-apimanagement/2017-03-01/swagger/apimproperties.json:143:6
>>  I have made the GetContract and CreateContract same
    arm-apimanagement/2017-03-01/swagger/apimsubscriptions.json:146:6
>> state is a required parameter in GetContract and NotRequired in CreateContract
    arm-apimanagement/2017-03-01/swagger/apimusers.json:146:6
>> groups is a parameter in the GetContract, but not present in CreateContract
    arm-apimanagement/2017-03-01/swagger/apimopenidconnectproviders.json:140:6 
>>  I have made the GetContract and CreateContract same

**Review by fearthecowboy — APPROVED** [review]

**Pull request merged**